### PR TITLE
Preview changes from typespec-next

### PR DESCRIPTION
**Pull request by mikeharder**
* Swagger ModelValidation, Swagger Avocado
  * Likely existing failures in main unrelated to this PR
* Swagger BreakingChanges

### CODE REVIEW

File: eng/pipelines/templates/steps/npm-install.yml [hunk 1]
```diff
@@ -11,8 +11,8 @@ steps:
   - script: npm --version --loglevel info
     displayName: npm --version
 
-  - script: npm ci
-    displayName: npm ci
+  - script: npm install --no-package-lock
```
Comment (markcowl) [inline eng/pipelines/templates/steps/npm-install.yml:14]: We don’t include this in the final pr, right?

File: specification/devopsinfrastructure/Microsoft.DevOpsInfrastructure/main.tsp [hunk 2]
```diff
@@ -353,6 +353,7 @@ model ResourceDetailsObject is ProxyResource<ResourceDetailsObjectProperties> {
   @segment("resources")
   @visibility("read")
   @path
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
```
Comment (markcowl) [inline specification/devopsinfrastructure/Microsoft.DevOpsInfrastructure/main.tsp:356]: I think we should probably suppress this rule unless this was checked with the service team

File: specification/networkanalytics/NetworkAnalytics.Management/main.tsp [hunk 3]
```diff
@@ -464,6 +465,7 @@ model DataProductsCatalog is ProxyResource<DataProductsCatalogProperties> {
   @key("dataProductsCatalogName")
   @segment("dataProductsCatalogs")
   @path
```
Comment (markcowl) [inline specification/networkanalytics/NetworkAnalytics.Management/main.tsp:467]: Again, without service input, we should suppress these

File: specification/playwrighttesting/PlaywrightTesting.Management/main.tsp [hunk 4]
```diff
@@ -112,6 +112,7 @@ model Quota is ProxyResource<QuotaProperties> {
   @key("quotaName")
   @path
   @segment("quotas")
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
```
Comment (markcowl) [inline specification/playwrighttesting/PlaywrightTesting.Management/main.tsp:115]: Suppress instead

File: specification/sphere/Sphere.Management/certificate.tsp [hunk 5]
```diff
@@ -24,6 +24,7 @@ model Certificate is ProxyResource<CertificateProperties> {
   @key("serialNumber")
   @segment("certificates")
   @visibility("read")
```
Comment (markcowl) [inline specification/sphere/Sphere.Management/certificate.tsp:26]: Suppress instead, as above

File: specification/sphere/Sphere.Management/image.tsp [hunk 6]
```diff
@@ -24,6 +24,7 @@ model Image is ProxyResource<ImageProperties> {
   @key("imageName")
   @segment("images")
   @visibility("read")
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
```
Comment (markcowl) [inline specification/sphere/Sphere.Management/image.tsp:27]: Suppress

File: specification/vmware/Microsoft.AVS/models.tsp [hunk 7]
```diff
@@ -1135,21 +1212,23 @@ model WorkloadNetwork is ProxyResource<WorkloadNetworkProperties> {
   @key("workloadNetworkName")
   @path
   @segment("workloadNetworks")
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
```
Comment (markcowl) [inline specification/vmware/Microsoft.AVS/models.tsp:1215]: Suppress

File: specification/vmware/Microsoft.AVS/models.tsp [hunk 8]
```diff
@@ -2100,6 +2187,7 @@ model IscsiPath is ProxyResource<IscsiPathProperties> {
   @key("iscsiPathName")
   @path
   @segment("iscsiPaths")
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
```
Comment (markcowl) [inline specification/vmware/Microsoft.AVS/models.tsp:2190]: Suppress

**Review by markcowl — CHANGES_REQUESTED** [review]
Would change the added pattern decorators to a rule suppression

File: eng/pipelines/templates/steps/npm-install.yml (hunk 1 above)
Comment (mikeharder) [reply eng/pipelines/templates/steps/npm-install.yml:14]: Correct, this is just a hack to avoid needing to update the lockfile in typespec-next.

**Pull request closed without merging**